app/faceanalysis/models.py

```diff
@@ -62,12 +74,60 @@ class Match(Base):  # type: ignore
     that_img = relationship('Image', foreign_keys=[that_img_id])
     __table_args__ = (UniqueConstraint(
         'this_img_id', 'that_img_id', name='_this_that_uc'),)
-# pylint: enable=too-few-public-methods
 
 
-def init_models(database_engine):
-    Base.metadata.create_all(database_engine)
+class FaceApiMapping(Base):  # type: ignore
+    __tablename__ = 'faceapimappings'
+
+    id = Column(Integer, primary_key=True)
+    img_id = Column(String(50), index=True, unique=True)
+    face_id = Column(String(50), index=True, unique=True)
+# pylint: enable=too-few-public-methods
 
 
-def delete_models(database_engine):
-    Base.metadata.drop_all(database_engine)
+class ImageStatusEnum(Enum):
+    finished_processing = 4
+    processing = 3
+    uploaded = 1
+    face_vector_computed = 5
+
+
+@lru_cache(maxsize=1)
+def _connect():
+    engine = create_engine(SQLALCHEMY_CONNECTION_STRING, pool_recycle=3600)
+    session_factory = sessionmaker(bind=engine)
+    return engine, session_factory
+
+
+def init_models():
+    engine, _ = _connect()
+    Base.metadata.create_all(engine)
+
+
+def delete_models():
+    engine, _ = _connect()
+    Base.metadata.drop_all(engine)
+
+
+# pylint: disable=broad-except
+@contextmanager
```

Comment:
This is such a good use of contextmanager!

Reply:
Custom context managers are a very nice way to clean up code. Definitely a powerful tool worth learning.